### PR TITLE
allow downstream apps or config-specific override to docs description

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request allows overriding the plugin description text in the plugin's python class, so that there can easily be per-config overrides as well as support for downstream apps to set custom text.

This was split out of #2706 to be applied to all plugins as well as the style guide.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by pllim — APPROVED** [review]
Seems uncontroversial enough. Thanks!

File: jdaviz/configs/cubeviz/plugins/slice/slice.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
   <j-tray-plugin
-    description='Select slice (or wavelength) of the cube to show in the image viewers and highlighted in the spectrum viewer.  The slice can also be changed interactively in the spectrum viewer by activating the slice tool.'
+    :description="docs_description || 'Select slice of the cube to show in the image viewers.  The slice can also be changed interactively in the spectrum viewer by activating the slice tool.'"
```
[the remark this answers is not in the document]
Comment (kecnry) [reply jdaviz/configs/cubeviz/plugins/slice/slice.vue]: now that #2706 was merged first, this should be fine as is.

**Review by gibsongreen — APPROVED** [review]
Everything looks straightforward, approved and ready to merge!

File: jdaviz/configs/default/plugins/markers/markers.vue [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <template>
   <j-tray-plugin
-    description='Create and export markers.  Press "m" with the cursor over a viewer to log the mouseover information.  To change the selected layer, click the layer cycler in the mouseover information section of the app-level toolbar.'
+    :description="docs_description || 'Create and export markers.  Press "m" with the cursor over a viewer to log the mouseover information.  To change the selected layer, click the layer cycler in the mouseover information section of the app-level toolbar.'"
```
Comment (pllim) [inline jdaviz/configs/default/plugins/markers/markers.vue:3]: BTW this broke the plugin but I fixed it directly on main via https://github.com/spacetelescope/jdaviz/commit/b58a5a44e7f8df42c02e519194d6967213ec125f . I was in the middle of something else and didn't feel like the overhead of JIRA + new PR.

Comment (kecnry) [reply jdaviz/configs/default/plugins/markers/markers.vue:3]: Woops, that one slipped by, thanks for the quick patch

**Pull request merged**